components/gateway/cmd/main.go

```diff
@@ -247,21 +245,23 @@ func fillJWTCredentials(cfg auditlog.OAuthConfig) clientcredentials.Config {
 	}
 }
 
-func initWorkers(workers chan bool, auditlogSvc proxy.AuditlogService, done chan bool, msgChannel chan proxy.AuditlogMessage, collector *metrics.AuditlogCollector) {
-	go func() {
+func initWorkers(ctx context.Context, workers chan bool, auditlogSvc proxy.AuditlogService, done chan bool, msgChannel chan proxy.AuditlogMessage, collector *metrics.AuditlogCollector) {
+	logger := log.C(ctx)
+
+	go func(log *logrus.Entry) {
```

Comment:
Why passing the logger to the goroutine is needed when it is available in the closure?

Reply:
Well spotted. It was not intended really.